nntrainer/layers/conv2d_layer.cpp

```diff
@@ -128,9 +128,19 @@ void Conv2DLayer::forwarding(sharedConstTensors in) {
     Tensor out = hidden_.getBatchSlice(b, 1);
     Tensor inSub = input_.getBatchSlice(b, 1);
 
-    status =
-      conv2d_gemm(filter_kernel.getData(), filter_kernel.getDim(), inSub,
-                  out_dim, stride, padding, out.getData(), out.length(), true);
+    std::vector<float> in_col;
```

Comment:
Please use Tensor instead of vector here to manage memory uniformly.

Reply:
Also, make this tensor a class variable so as to avoid allocaiton and deallocation with every forwarding.

Reply:
This is just temporal usage. So do you think it should be there all the time when forwarding and backwarding?

Reply:
Temporal usage will allocate and deallocate memory every iteration creating unnecessary overhead.
The allocation should be just done once.
This might increase peak memory (its possible that this allocation is already counted in peak memory) but increase performance.